### PR TITLE
Javascript Capstone Project

### PR DESCRIPTION
**Pull request by Lul-Abdifan**
### Summary
-  Build single-page application   :white_check_mark:
-  Use API documentation  :white_check_mark:
    -  likes, comments, and meal API are used.
-  Send and receive data from an API.  :white_check_mark:
-  Use JavaScript to manipulate DOM elements :white_check_mark:
-  Write unit tests   :white_check_mark: 
    - test is done for the comment counter on the PopUp page
    - test is done for the item count on the front page
-  Perform a code review for a team member.  :white_check_mark:
   - We did a professional code review practices
   -  Manage to fix occurred conflicts together 
   <hr>
 ### Files and Folders
-  Add modules folder for all functionalities
- Add style.scss 
  -  Update the styles to the responsiveness of all versions
- Add __test__ folder for testing
- Update README

If you have any questions, We are here @abeltsew   @Lul-Abdifan. 
We appreciate your understanding and patience. :desktop_computer: Happy Coding :desktop_computer:

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  fetch(mainAPI)
+    .then((response) => response.json())
+    .then((data) => {
+      const slicedData = data.meals.slice(0, 24);
+
+      slicedData.forEach((list) => {
+        renderMeal(list);
+        renderLike(list.idMeal, likes);
+      });
+      itemCounter.innerHTML = `Food(${slicedData.length})`;
```
Comment (amejid) [inline src/index.js:21]: - [ ] You did a great job on almost all functionalities. :+1: Let's refactor the counter to achieve the project requirements. Here you are using the incoming data from the API to render the number of items. According to the requirement, we should create a function to count the number of items displayed on the screen. Kindly create a function to count the HTML nodes displayed on the screen and you can use the output of the function to render the number of foods displayed on the screen. ✍️
Please implement the counter function as a separate module to make it easier to cover it with the test.

**Project requirements**
Screenshot from 2023-04-19 19-44-45

File: src/modules/renderComment.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
+import { baseUrl, involvementApi } from '../../config/keys.js';
+import getItemComments from './getCommentItems.js';
+
+let commentList = [];
+
+export const updateCommentCount = (comment, commentsHeader, add) => {
+  if (!comment.error && add !== true) {
+    commentsHeader.innerHTML = `comments (${comment.length})`;
```
Comment (amejid) [inline src/modules/renderComment.js:8]: - [ ] The same thing is happening here, you are utilizing the incoming data from the API to display the comments count. Please create a separate function as discussed earlier to handle the counting. 🥸️

File: __test__/itemCounter.test.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import itemCounters from '../src/updatedItemCounter.js';
+
+describe('itemCounters', () => {
```
Comment (amejid) [inline __test__/itemCounter.test.js:7]: - [ ] Since we discussed refactoring the counter function, the test should cover the function soon to be created. Please make sure you are covering all possible edge cases. One of them is when there are no foods displayed on the screen the counter function should return 0. ✍️

File: __test__/commentCounter.test.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { updateCommentCount } from '../src/modules/renderComment.js';
+
+describe('Should render the exact comments count', () => {
```
Comment (amejid) [inline __test__/commentCounter.test.js:7]: - [ ] Please do the same thing here as the itemCounter test. 🥸️

File: src/modules/getCommentItems.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { baseUrl, involvementApi } from '../../config/keys.js';
+
+export default async (item) => {
+  const response = await fetch(
+    // eslint-disable-next-line
```
Comment (amejid) [inline src/modules/getCommentItems.js:5]: - [ ] Enforcing linters in our project helps us write clean and consistent code. Kindly remove the line you used to disable linter checks. 😎️

File: src/modules/renderComment.js [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+  const commments = document.createElement('div');
+  const commentsHeader = document.createElement('h2');
+
+  commments.appendChild(commentsHeader);
+
+  commentUI.appendChild(commments);
+
+  getItemComments(item).then((comment) => {
+    // eslint-disable-next-line
```
Comment (amejid) [inline src/modules/renderComment.js:60]: - [ ] Same thing here. 🤝️

File: src/modules/renderComment.js [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+    !comment.error ? (commentList = comment) : [];
+    updateCommentCount(comment, commentsHeader);
+
+    comment?.forEach((comment) => {
+      const p = document.createElement('p');
+      const { creation_date: date, username, comment: commentText } = comment;
+      // eslint-disable-next-line
```
Comment (amejid) [inline src/modules/renderComment.js:67]: - [ ] Here. 🤝️

**Review by amejid — CHANGES_REQUESTED** [review]
Hi @abeltsew and @Lul-Abdifan, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- Linter checks passing :heavy_check_mark: 
- Correct workflow :heavy_check_mark: 
- Descriptive PR :heavy_check_mark: 
- Professional README :heavy_check_mark: 
- Good UI design :heavy_check_mark: 
- Functionalities working as expected :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Lul-Abdifan** [comment]
## Required Changes
#### We did the following changes
-  Create a function to count the number of items displayed on the screen.
-  implement the counter function as a separate module to make it easier to cover it with the test display of the comments count.
- Update the item Counter function to all possible edge cases
-  Remove the unnecessary lines of code
<hr>We appreciate your understanding. Have a nice time with coding
:smile: :smile:

**Review by fobadara — APPROVED** [review]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Linters are passing
✅ Good job adding a test that checks if there are no food present

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Lul-Abdifan** [comment]
@fobadara, We appreciate your time and your approval. Thank you
Enjoy your time coding with your fingers! :smile: :smile:
